driver-core/src/test/unit/com/mongodb/connection/CommandMessageSpecification.groovy

```diff
@@ -152,8 +152,92 @@ class CommandMessageSpecification extends Specification {
         ]
     }
 
-    def 'should respect the message settings limits'() {
+    def 'should respect the max message size'() {
```

Comment:
This test is a bit better as it can assert explicitly on the message lengths for each split.